week3/createContentTrainingData.py

```diff
@@ -3,10 +3,23 @@
 import random
 import xml.etree.ElementTree as ET
 from pathlib import Path
+from nltk.stem import SnowballStemmer
+from collections import Counter
+import re
+
+stemmer = SnowballStemmer('english')
+
+
+def transform_target(child) -> str:
+    # print(child.find('categoryPath')[len(child.find('categoryPath')) - 1][0].text, [child.find('categoryPath')[i][0].text for i in range(len(child.find('categoryPath')))])
+    return child.find('categoryPath')[len(child.find('categoryPath')) - 1][0].text
+    # return child.find('categoryPath')[min(3, len(child.find('categoryPath')) - 1)][0].text
+
+
+def transform_name(product_name: str) -> str:
```

Comment:
nice! You can also remove punctuation using `string`

`import string`

`translator = str.maketrans("", "", string.punctuation)
 product = product_name.translate(translator)`

Reply:
good to know ! first time I hear about translate/translator, and string.punctuation also !  really interesting, thanks !

Reply:
not sure if it's the best way though! Saw some use `[word for word if word.isalpha()]` 

also used `word_tokenize` function in `nltk` but I assume product.split() would achieve similar results